include/gunrock/algorithms/mst.hxx

```diff
@@ -0,0 +1,302 @@
+      // later beaten by lower weights.
+      auto source = G.get_source_vertex(e);
+      auto dest = G.get_destination_vertex(e);
+      auto weight = G.get_edge_weight(e);
+
+      if (source < dest && roots[source] != roots[dest]) {
+        if (weight == min_weights[roots[source]]) {
+          math::atomic::min(&(min_neighbors[roots[source]]), e);
+        }
+        if (weight == min_weights[roots[dest]]) {
+          math::atomic::min(&(min_neighbors[roots[dest]]), e);
+        }
+      }
+    };
+
+    auto add_to_mst = [G, roots, mst_weight, mst_edges, super_vertices,
+                       min_neighbors, min_weights, new_roots, not_decremented,
+                       mst] __host__
+                      __device__(vertex_t const& v) -> void {
+      // Add weights to MST. To prevent duplicate edges, check that either
+      // the source vertex index is
+      // less than the destination vertex index or that the edge with the source
+      // and destination flipped is not included.
+      // Combine super vertices by setting the root of the super vertex to the
+      // root of the destination. Use `roots` to check old roots and update
+      // roots in `new_roots`.
+      if (min_weights[v] != std::numeric_limits<weight_t>::max()) {
+        auto e = min_neighbors[v];
+        auto source = G.get_source_vertex(e);
+        auto dest = G.get_destination_vertex(e);
+        auto weight = G.get_edge_weight(e);
+
+        if (roots[source] != v) {
+          auto temp = source;
+          source = dest;
+          dest = temp;
+        }
+
+        if (source < dest || min_neighbors[roots[dest]] != e) {
+          // For large graphs with float weights, there may be slight variance
+          // in the final MST weight due to atomic adds and the amount of
+          // precision loss depending on the order of adds.
+          not_decremented[0] = false;
+          math::atomic::add(&mst_weight[0], weight);
+          math::atomic::add(&mst_edges[0], 1);
+          math::atomic::add(&super_vertices[0], -1);
+          math::atomic::exch(&new_roots[v], new_roots[dest]);
+        }
+      }
+    };
+
+    auto jump_pointers_parallel =
+        [new_roots] __host__ __device__(vertex_t const& v) -> void {
+      // Update the root of each vertex. When adding an edge to the MST, we
+      // update the source's root's root to the destination's root.
+      // However, any vertex that had the source's root as its root would not
+      // be updated. We must jump from root to root until the current vertex
+      // and root are equal to find the new roots.
+      vertex_t u = new_roots[v];
+      while (new_roots[u] != u) {
+        u = new_roots[u];
+      }
+      new_roots[v] = u;
+      return;
+    };
+
+    auto in_frontier = &(this->frontiers[0]);
+    auto out_frontier = &(this->frontiers[1]);
+
+    // Execute filter operator to get min weights
+    operators::filter::execute<operators::filter_algorithm_t::remove>(
```

Comment:
`filter_algorithm_t::bypass` maybe faster, you can play around with this to see.